### ISSUE TITLE
[Campaign] [Delfador's Memoirs] Update the substandard Quality of Scenario Maps

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#7479]
The map files for this one in particular look very outdated, substandard, deficient in quality, unsatisfactory and not up to par with the maps in the other mainline campaigns. 

The maps should be enhanced to higher grade without impacting the scenarios too much.

**Comment by stevecotton** [comment]
As this is labelled a "Good first issue", I feel we should decide right now which maps / scenarios would definitely be kept in any other planned changes for the campaign, so that we don't end up accepting a new contributor and then shortly afterwards removing their first big contribution.

IMO (although not my decision), I'd suggest the Land of the Dead should be deleted rather than improved.

**Comment by knyghtmare** [comment]
> As this is labelled a "Good first issue", I feel we should decide right now which maps / scenarios would definitely be kept in any other planned changes for the campaign.

Not sure if the rework would happen or not. At least not in the current decade I guess. Thus, all the maps should be improved to higher grade.

> IMO (although not my decision), I'd suggest the Land of the Dead should be deleted rather than improved.

> In some aspects, yes. But I think DM really needs the Land of the Dead part. It's one of the only things it has going for it.

One of the unique things (not many to begin with) about it is the Land of the Dead so I'd be more in favour of expanding that chapter instead of the meager 2 brief scenarios.

The other unique is shifting recall lists (Loyalists, Ghosts&Ghouls, Rebels). Take those away, and DM is just another AOI.

**Comment by cooljeanius** [comment]
IMO this isn't really actionable unless you specify more precisely what exactly you're looking for here. "Quality" is a subjective measure, and different people have different standards for it when it comes to the aesthetics of map design. Some people prefer simplicity, some people prefer complexity. Some people prefer retro styles, others prefer more modern ones. Listing what exactly your criteria for better maps are would help people meet those criteria more easily.

**Comment by knyghtmare** [comment]
> you specify more precisely what exactly you're looking for here

1. More than 10 terrain types used. (some maps have just 5 or even less...)
2. Map is actually visually appealing to look at. Looks too plain (compare with Liberty/DiD/TSG maps for Quality comparison)
3. Map appearance uses the campaign's lore properly. Land of the Dead map look like a too simplistic compared to what the campaign is desperately trying to conceptualize.
4. Uses embellishments. Dont think any map in DM does...

**Comment by Wedge009** [comment]
> "Quality" is a subjective measure...

My thoughts too. There are a lot of these 'improve quality' issues that I wonder would fit better in the forums discussions. Subjective issues like these (and #7106 and #7365) are bit harder to tackle than an issue about a known bug or a specific feature request. On top of those you also have people remarking about consistency - or lack of it - between scenarios (#7126, #7127, #7137). What's a campaign designer to do?

**Comment by szymor** [comment]
In 1.18.3 the scenario Prince of Wesnoth from Delfador's Memoirs stalls the game - after briefing and showing a cross on the map there is no possibility to start a scenario as it hangs. Pressing Skip gives the similar result.

**Comment by cooljeanius** [comment]
> In 1.18.3 the scenario Prince of Wesnoth from Delfador's Memoirs stalls the game - after briefing and showing a cross on the map there is no possibility to start a scenario as it hangs. Pressing Skip gives the similar result.

I don't think this is related to the maps?

**Comment by szymor** [comment]
I thought it was as it is related to a WML scripting issue for the map of this scenario, but I already solved it. It seems you need to wait really long, but in the end the game will continue. The Lua script named DELFADOR_JOINS_ELVES takes much time to execute.

**Comment by gfgtdf** [comment]
> I thought it was as it is related to a WML scripting issue for the map of this scenario, but I already solved it. It seems you need to wait really long, but in the end the game will continue. The Lua script named DELFADOR_JOINS_ELVES takes much time to execute.

This sounds like a bug, that script doennt do anythign that should take time

**Comment by gfgtdf** [comment]
maybe it tries to animate the advancement in unit:advance()

**Comment by Wedge009** [comment]
> In 1.18.3 the scenario Prince of Wesnoth from Delfador's Memoirs stalls the game - after briefing and showing a cross on the map there is no possibility to start a scenario as it hangs. Pressing Skip gives the similar result.

This should be reported as a separate issue.

**Comment by szymor** [comment]
> maybe it tries to animate the advancement in unit:advance()

Exactly, it spends much time on it. It is visible if you change the event type from prestart to start. If you comment out the Lua code within the macro, the scenario will launch instantly.

**Comment by knyghtmare** [comment]
Please move "out of issue" scope discussions to their own bug report...

**Comment by gfgtdf** [comment]
> Exactly, it spends much time on it. It is visible if you change the event type from prestart to start. If you comment out the Lua code within the macro, the scenario will launch instantly.

I fixed this in for now in https://github.com/wesnoth/wesnoth/commit/e549b03412e3d735039f98c1d66f68b1d55b2eff (master only)

> I was not able to replicate this in 1.18.3 release. But often with issues to do with 'slow' processing, I cannot replicate.

I probably depends on how many units you have and maybe also on whether you have accelerated speed enabled

But yeah it's kinda off topic here.

**Comment by knyghtmare** [comment]
Closed by #9726
